### PR TITLE
[RHPAM-2175] Remove controllerUser and executionUser

### PR DESCRIPTION
**Pull request by ruromero**
Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

Fix https://issues.jboss.org/browse/RHPAM-2175

### CODE REVIEW

**Comment by ruromero** [comment]
@sutaakar @rhtevan please review as well
/assign @tchughesiv

**Comment by ruromero** [comment]
/hold
depends on https://github.com/jboss-container-images/jboss-kie-modules/pull/334

File: deploy/crds/kieapp.crd.yaml [hunk 1]
```diff
@@ -83,12 +83,6 @@ spec:
                     amqClusterPassword:
                       type: string
                       description: The password to use for amq cluster user.
-                    controllerPassword:
-                      type: string
-                      description: The password to use for the controllerUser.
-                    serverPassword:
-                      type: string
-                      description: The password to use for the executionUser.
```
Comment (tchughesiv) [inline deploy/crds/kieapp.crd.yaml]: I'm not sure if we can remove these items from the CRD until versions 7.5.x are gone from available/deployable versions in operator. Have you tested if those versions will deploy properly still?

Comment (tchughesiv) [reply deploy/crds/kieapp.crd.yaml]: unless you can also change those version config files as well?

**Comment by rhtevan** [comment]
@ruromero  Using kie_admin_user credential in place of kie_server_user and kie_server_controller_user brings a minor conflict in that existing scripts will try to add the same user twice which will emit error messages. It would better to generate a new username/passwd pair instead of reusing admin user credential. wdyt? 

```
ezhang@devf jboss-kie-modules]$ grep -lR -e 'add_kie_server_user' --include=*.sh
jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-security.sh
jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh
[ezhang@devf jboss-kie-modules]$ grep -lR -e 'add_kie_server_controller_user' --include=*.sh
jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-security.sh
jboss-kie-controller/added/launch/jboss-kie-controller.sh
jboss-kie-workbench/added/launch/jboss-kie-workbench.sh
[ezhang@devf jboss-kie-modules]$ grep -lR -e 'add_kie_admin_user' --include=*.sh
jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-security.sh
jboss-kie-workbench/added/launch/jboss-kie-workbench.sh
jboss-kie-optaweb-common/added/launch/jboss-kie-optaweb-common.sh
jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh
```

**Comment by ruromero** [comment]
@rhtevan that is not a problem, the jboss-cli will be called twice with the same user/pwd tuple and nothing will happen.

File: deploy/crds/kieapp.crd.yaml (hunk 1 above)
Comment (ruromero) [reply deploy/crds/kieapp.crd.yaml]: Items have been removed from CRD but kept in the Go structs to prevent conflicts from 7.5.x.

**Comment by ruromero** [comment]
@tchughesiv let me know if 7.4 check should be there at all.
Thanks

**Comment by tchughesiv** [comment]
@ruromero according to jira this is targeted for 7.7.0? if so, we have to wait until master advances to that version... currently at 7.6.0

**Comment by spolti** [comment]
@tchughesiv  correct.

**Comment by errantepiphany** [comment]
@tchughesiv I know this is already on "/hold" since it is targeted for 7.7, however please leave it "held" until after some of us get to discuss it further. I'm not confident this is the approach we want to take yet until I get to have a video call with @rhtevan , @ruromero (and hopefully @spolti ) so I understand it better. Thanks!

**Comment by spolti** [comment]
count with me.

**Comment by tchughesiv** [comment]
please rebase and update to use 7.7.0 configs

**Comment by tchughesiv** [comment]
@ruromero probably best to wait for https://github.com/kiegroup/kie-cloud-operator/pull/309 to merge before rebasing

**Comment by ruromero** [comment]
@tchughesiv rebased

**Pull request merged**